File: 11-06-2025_Remove_the_ball.java
class Solution {
    public int findLength(List<Integer> color, List<Integer> radius) {
        Stack<int[]> stack = new Stack<>();

        for (int i = 0; i < color.size(); i++) {
            int[] curr = {color.get(i), radius.get(i)};
            if (!stack.isEmpty()) {
                int[] top = stack.peek();
                if (top[0] == curr[0] && top[1] == curr[1]) {
                    stack.pop();
                    continue;
                }
            }
            stack.push(curr);
        }

        return stack.size();
    }
}

File: 11-06-2025_Remove_the_ball.cpp
class Solution {
  public:
    int findLength(vector<int> &color, vector<int> &radius)
    {
        int n=color.size();
        int ans=0;
        int i=0;
        
        stack<pair<int,int>> s;
        
        for(;i<n;++i)
        {
            pair<int,int> curr={color[i],radius[i]};
            if(!s.empty() && s.top()==curr)
                s.pop();
            else
                s.push(curr);
        }
        return s.size();
    }
};